### PR TITLE
Add comdb2_functions table and autocomplete ability for them

### PR DESCRIPTION
**Pull request by adizaimi**
Add new system table `comdb2_functions` table.
Add ability to autocomplete all sqlite and comdb2 functions from
cdb2sql command line.

Signed-off-by: Adi Zaimi <azaimi@bloomberg.net>

### CODE REVIEW

File: sqlite/ext/comdb2/functions.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include <stdlib.h>
```
Comment (nirbhayc) [inline sqlite/ext/comdb2/functions.c:1]: License missing.

File: sqlite/ext/misc/completion.c [hunk 2]
```diff
@@ -332,6 +339,10 @@ static int completionNext(sqlite3_vtab_cursor *cur){
           pCur->szRow = strlen(pCur->zCurrentRow);
         }
         iCol = -1;
+         */
```
Comment (nirbhayc) [inline sqlite/ext/misc/completion.c:342]: [Optional] I'd remove instead of commenting them out.

File: sqlite/ext/misc/completion.c [hunk 3]
```diff
@@ -309,7 +310,12 @@ static int completionNext(sqlite3_vtab_cursor *cur){
       }
 #if defined(SQLITE_BUILDING_FOR_COMDB2)
       case COMPLETION_FUNCTIONS: {
-        /* NOTE: Please keep this list of functions sorted. */
+        if( pCur->pStmt==0 ){
+          sqlite3_prepare_v2(pCur->db, 
+                  "SELECT name FROM comdb2_functions",
```
Comment (nirbhayc) [inline sqlite/ext/misc/completion.c:315]: order by?

File: tests/auth.test/t09.expected [hunk 4]
```diff
@@ -227,6 +227,7 @@
 (candidate='comdb2_cron_events')
 (candidate='comdb2_cron_schedulers')
 (candidate='comdb2_fingerprints')
+(candidate='comdb2_functions')
```
Comment (nirbhayc) [inline tests/auth.test/t09.expected:230]: Please also update: https://bloomberg.github.io/comdb2/system_tables.html

File: sqlite/ext/comdb2/functions.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+#include "comdb2.h"
+#include "comdb2systbl.h"
+#include "comdb2systblInt.h"
+#include "sql.h"
+#include "ezsystables.h"
+#include "cdb2api.h"
+
+static int get_functions(void **data, int *npoints)
+{
+    void get_sqlite_functions(void **data, int *tot);
```
Comment (mponomar) [inline sqlite/ext/comdb2/functions.c:14]: Move declaration to header?

Comment (adizaimi) [reply sqlite/ext/comdb2/functions.c:14]: will do.

File: sqlite/ext/comdb2/functions.c (hunk 1 above)
Comment (adizaimi) [reply sqlite/ext/comdb2/functions.c:1]: adding. also to all the other files in the directory : )

File: sqlite/ext/misc/completion.c (hunk 2 above)
Comment (adizaimi) [reply sqlite/ext/misc/completion.c:342]: creating a new ifdef -- it is good to have how to add custom list for future reference.

File: tests/auth.test/t09.expected (hunk 4 above)
Comment (adizaimi) [reply tests/auth.test/t09.expected:230]: ok.

**Pull request merged**